winlogbeat/eventlog/wineventlog.go

```diff
@@ -113,8 +114,33 @@ type query struct {
 // any problems or nil.
 func (c *winEventLogConfig) Validate() error {
 	var errs multierror.Errors
-	if c.Name == "" {
-		errs = append(errs, fmt.Errorf("event log is missing a 'name'"))
+
+	if c.XMLQuery != "" {
+		if c.ID == "" {
+			errs = append(errs, fmt.Errorf("event log is missing an 'id'"))
+		}
+
+		var tmp interface{}
+		if err := xml.Unmarshal([]byte(c.XMLQuery), &tmp); err != nil {
```

Comment:
This is sort of surprising use of an `interface{}` value. Maybe `xml.Unmarshal([]byte(c.XMLQuery), new(struct{}))` to signal that the only result we care about here is the error.

Reply:
All we care about here is if the XML is valid or not, so we just need the error. Agreed that empty struct is a better fit. If we ever wanted to check for the validity of the query itself, we'd need a Windows function to do that, but I'm not aware of one.